crates/providers/src/heart.rs

```diff
@@ -16,8 +19,150 @@ use tokio::{
+}
+
+impl<N: Network, T: Transport + Clone, P: Provider<N, T>> PendingTransactionConfig<N, T, P> {
+    /// Creates a new pending transaction configuration.
+    pub fn new(provider: P, tx_hash: B256) -> Self {
+        Self::from_inner(provider, PendingTransactionConfigInner::new(tx_hash))
+    }
+
+    /// Creates a new pending transaction configuration.
+    pub fn from_inner(provider: P, inner: PendingTransactionConfigInner) -> Self {
+        Self { inner, provider, _phantom: PhantomData }
+    }
+
+    /// Returns the inner configuration.
+    pub fn inner(&self) -> &PendingTransactionConfigInner {
+        &self.inner
+    }
+
+    /// Consumes this configuration, returning the inner configuration.
+    pub fn into_inner(self) -> PendingTransactionConfigInner {
+        self.inner
+    }
+
+    /// Returns the provider.
+    pub fn provider(&self) -> &P {
+        &self.provider
+    }
+
+    /// Consumes this configuration, returning the provider and the inner configuration.
+    pub fn split(self) -> (P, PendingTransactionConfigInner) {
+        (self.provider, self.inner)
+    }
+
+    /// Returns the transaction hash.
+    pub fn tx_hash(&self) -> &B256 {
+        self.inner.tx_hash()
+    }
+
+    /// Sets the transaction hash.
+    pub fn set_tx_hash(&mut self, tx_hash: B256) {
+        self.inner.set_tx_hash(tx_hash);
+    }
+
+    /// Sets the transaction hash.
+    pub fn with_tx_hash(mut self, tx_hash: B256) -> Self {
+        self.set_tx_hash(tx_hash);
+        self
+    }
+
+    /// Returns the number of confirmations to wait for.
+    pub fn confirmations(&self) -> u64 {
+        self.inner.confirmations()
+    }
+
+    /// Sets the number of confirmations to wait for.
+    pub fn set_confirmations(&mut self, confirmations: u64) {
+        self.inner.set_confirmations(confirmations);
+    }
+
+    /// Sets the number of confirmations to wait for.
+    pub fn with_confirmations(mut self, confirmations: u64) -> Self {
+        self.set_confirmations(confirmations);
+        self
+    }
+
+    /// Returns the timeout.
+    pub fn timeout(&self) -> Option<Duration> {
+        self.inner.timeout()
+    }
+
+    /// Sets the timeout.
+    pub fn set_timeout(&mut self, timeout: Option<Duration>) {
+        self.inner.set_timeout(timeout);
+    }
+
+    /// Sets the timeout.
+    pub fn with_timeout(mut self, timeout: Option<Duration>) -> Self {
+        self.set_timeout(timeout);
+        self
+    }
+
+    /// Registers the watching configuration with the provider.
+    ///
+    /// This does not wait for the transaction to be confirmed, but returns a [`PendingTransaction`]
+    /// that can be awaited at a later moment.
+    ///
+    /// See:
+    /// - [`watch`](Self::watch) for watching the transaction without fetching the receipt.
+    /// - [`get_receipt`](Self::get_receipt) for fetching the receipt after the transaction has been
+    ///   confirmed.
+    pub async fn register(self) -> TransportResult<PendingTransaction> {
```

Comment:
register -> PendingTransaction
wait -> PendingTransaction.await -> B256
get_receipt -> PendingTransaction.await.await -> Option<Receipt> (turn option into an error?)

Not super sure about names, open to suggestions to the API as well